wdl-analysis/src/handlers/document_symbol.rs

```diff
@@ -0,0 +1,380 @@
+                children.push(scatter_to_symbol(uri, &scatter, lines)?)
+            }
+            _ => {}
+        }
+    }
+
+    Ok(DocumentSymbol {
+        name: workflow.name().text().to_string(),
+        detail: Some("workflow".to_string()),
+        kind: SymbolKind::MODULE,
+        range: common::location_from_span(uri, workflow.span(), lines)?.range,
+        selection_range: common::location_from_span(uri, workflow.name().span(), lines)?.range,
+        children: Some(children),
+        tags: None,
+        #[allow(deprecated)]
+        deprecated: None,
+    })
+}
+
+/// Converts a [`TaskDefinition`] to a [`DocumentSymbol`].
+fn task_to_symbol(
+    uri: &Url,
+    task: &TaskDefinition,
+    lines: &std::sync::Arc<line_index::LineIndex>,
+) -> Result<DocumentSymbol> {
+    let mut children = Vec::new();
+
+    if let Some(input_section) = task.input() {
+        children.extend(input_section_to_symbols(uri, &input_section, lines)?);
+    }
+
+    if let Some(output_section) = task.output() {
+        children.extend(output_section_to_symbols(uri, &output_section, lines)?);
+    }
+
+    for decl in task.declarations() {
+        children.push(bound_decl_to_symbol(uri, &decl, lines)?);
+    }
+
+    Ok(DocumentSymbol {
+        name: task.name().text().to_string(),
+        detail: Some("task".to_string()),
+        kind: SymbolKind::FUNCTION,
+        range: common::location_from_span(uri, task.span(), lines)?.range,
+        selection_range: common::location_from_span(uri, task.name().span(), lines)?.range,
+        children: Some(children),
+        tags: None,
+        #[allow(deprecated)]
+        deprecated: None,
+    })
+}
+
+/// Converts a [`StructDefinition`] to a [`DocumentSymbol`].
+fn struct_to_symbol(
+    uri: &Url,
+    s: &StructDefinition,
+    lines: &std::sync::Arc<line_index::LineIndex>,
+) -> Result<DocumentSymbol> {
+    let mut children = Vec::new();
+
+    for member in s.members() {
+        children.push(unbound_decl_to_symbol(uri, &member, lines)?);
+    }
+
+    Ok(DocumentSymbol {
+        name: s.name().text().to_string(),
+        detail: Some("struct".to_string()),
+        kind: SymbolKind::STRUCT,
+        range: common::location_from_span(uri, s.span(), lines)?.range,
+        selection_range: common::location_from_span(uri, s.name().span(), lines)?.range,
+        children: Some(children),
+        tags: None,
+        #[allow(deprecated)]
+        deprecated: None,
+    })
+}
+
+/// Converts an [`InputSection`] to a [`Vec<DocumentSymbol>`.]
+fn input_section_to_symbols(
+    uri: &Url,
+    section: &InputSection,
+    lines: &std::sync::Arc<line_index::LineIndex>,
+) -> Result<Vec<DocumentSymbol>> {
+    let mut symbols = Vec::new();
+    for decl in section.declarations() {
+        symbols.push(decl_to_symbol(uri, &decl, lines)?);
+    }
+    Ok(symbols)
+}
+
+/// Converts an [`OutputSection`] to a [`Vec<DocumentSymbol>`.]
+fn output_section_to_symbols(
```

Comment:
I wonder if it would make sense to group input and output declarations as children under
an `input` and `output` object (or a more suitable `SymbolKind`)?

Reply:
Maybe `SymbolKind::NAMESPACE`?